torch/_dynamo/decorators.py

```diff
@@ -458,6 +458,25 @@ def maybe_mark_dynamic(t, index):
         maybe_mark_dynamic(t, i)
 
 
+def mark_static_nn_module(mod_class):
```

Comment:
More commentary.

Today's world - Consider user defined nn module int attributes automatic dynamic. Use `mark_static_nn_module` to make them static if you want.

Alternative work - Consider all int attributes of any user defined nn module class static. And then introduce an API - `torch._dynamo.mark_nn_module_attribute_dynamic`. The default being static is worrying if users have counter in their model which is updated in each forward invocation.

Reply:
cc @ezyang @bdhirsh @zou3519